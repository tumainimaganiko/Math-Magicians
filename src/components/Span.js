import PropTypes from 'prop-types';

export default function Span({ text, color }) {
  return (
    <span className={color}>{text}</span>
  );
}

Span.propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
};
