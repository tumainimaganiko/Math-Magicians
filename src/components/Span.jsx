import PropTypes from 'prop-types';
import calculate from '../logic/calculate';

export default function Span({
  text, color, obj, setObj,
}) {
  function handleClick(valueBtn) {
    setObj(calculate(obj, valueBtn));
  }
  return (
    <button onClick={() => handleClick(text)} type="button" className={color}>
      {text}
    </button>
  );
}

Span.propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  obj: PropTypes.oneOfType([
    PropTypes.shape({
      total: PropTypes.string,
      next: PropTypes.string,
      operation: PropTypes.string,
    }),
    PropTypes.oneOf([null]),
  ]).isRequired,
  setObj: PropTypes.func.isRequired,
};
