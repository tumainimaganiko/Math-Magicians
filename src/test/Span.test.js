import renderer from 'react-test-renderer';
import Span from 'components/Span';

it('Be a button', () => {
    const obj = {
        total: null,
        next: null,
        operation: null,
    }
    const tree = renderer.create(<Span obj={obj} setObj={() => obj} text="AC" color="white" />).toJSON();
    expect(tree).toMatchSnapshot();
})