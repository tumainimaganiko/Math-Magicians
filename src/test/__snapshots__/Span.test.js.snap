// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Be a button 1`] = `
<button
  className="white"
  onClick={[Function]}
  type="button"
>
  AC
</button>
`;
